integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java

```diff
@@ -701,13 +706,14 @@ private int releaseInternal(String path, FuseFileInfo fi) {
     OpenFileEntry oe;
     final long fd = fi.fh.get();
     oe = mOpenFiles.getFirstByField(ID_INDEX, fd);
-    mOpenFiles.remove(oe);
-    if (oe == null) {
+    if (oe == null || !mOpenFiles.remove(oe)) {
       LOG.error("Cannot find fd for {} in table", path);
       return -ErrorCodes.EBADFD();
     }
     try {
-      oe.close();
+      synchronized (oe) {
+        oe.close();
```

Comment:
Do we need to check here that `oe` is still in the index?

Reply:
What about this?

Reply:
I think we don't need to check oe still in the index here, even, we can ensure that oe is not in the index certainly. Because `mOpenFiles.remove(oe)` must return true, otherwise, it should return `-ErrorCodes.EBADFD()`.